app/src/main.js

```diff
@@ -7,33 +7,57 @@ const serviceConfig = require('./config/service');
 const smmtConfigLoader = require('./config/smmtConfigurationLoader');
 const smmtClientFactory = require('./smmt/client');
 const loggerFactory = require('./logger/createLogger');
+const responseCode = require('./smmt/responseCode').code;
 
 const app = express();
 app.disable('x-powered-by');
 const traceHeaderName = 'x-mot-trace-id';
 
+function processSmmtResponse(make, vin, recall, res, logger) {
+  if (recall.success) {
+    logger.info({ context: { make, vin, recall } }, 'Recall fetched successfully.');
+    res.status(200)
+      .send({
+        status_description: recall.description,
+        vin_recall_status: recall.status,
+        last_update: recall.lastUpdate,
+      });
+  } else if (recall.actionCode === responseCode.smmtInvalidMarque) {
+    const context = {
```

Comment:
there is quite a bit of code replication in those else-ifs. Is it feasible to extract the content of those into a parametrised function?

Reply:
discussed f2f